record-editor.py

```diff
@@ -265,30 +321,31 @@ def create_models( self ):
         # NOTE: we keep photo id in the model so we can pull it from our
         #       selection and access the record's data.
         #
-        self.photosModel = QStandardItemModel( 0, 3, self )
+        self.photosModel = QStandardItemModel( len(self.photos), self.NUM_COLUMNS, self )
 
-        self.photosModel.setHeaderData( 0, Qt.Horizontal, "Photo ID" )
-        self.photosModel.setHeaderData( 1, Qt.Horizontal, "File Path" )
-        self.photosModel.setHeaderData( 2, Qt.Horizontal, "State" )
+        self.photosModel.setHeaderData( self.ID_COLUMN, Qt.Horizontal, "Photo ID" )
+        self.photosModel.setHeaderData( self.PATH_COLUMN, Qt.Horizontal, "File Path" )
+        self.photosModel.setHeaderData( self.STATE_COLUMN, Qt.Horizontal, "State" )
 
         # walk through each of the photo records and insert a new item at the
         # beginning of the model's list.
-        #
-        # XXX: there has got to be a better way to insert things.  the
-        #      QAbstractItemModel doesn't have an insert that grows.
-        #      inserting QStandardItems explicitly turns our id into an
-        #      empty string.  calling reversed() on the photos does not
-        #      actually do what we want.
-        for photo in self.photos:
-            self.photosModel.insertRow( 0 )
-            self.photosModel.setData( self.photosModel.index( 0, 0 ), photo["id"] )
-            self.photosModel.setData( self.photosModel.index( 0, 1 ), photo["filename"] )
-            self.photosModel.setData( self.photosModel.index( 0, 2 ), photo["state"] )
+        for index, photo in enumerate(self.photos):
+            id_item       = QStandardItem( photo["id"] )
+            filename_item = QStandardItem( photo["filename"] )
+            state_item    = QStandardItem( photo["state"] )
+
+            id_item.setData( int( photo["id"] ) )
+            filename_item.setData( photo["filename"] )
+            state_item.setData( photo["state"] )
+
+            self.photosModel.setItem( index, self.ID_COLUMN, id_item )
+            self.photosModel.setItem( index, self.PATH_COLUMN, filename_item )
+            self.photosModel.setItem( index, self.STATE_COLUMN, state_item )
 
         # create the proxy model for filtering our data based on record
         # processing state.
         self.proxyPhotosModel = QSortFilterProxyModel()
-        self.proxyPhotosModel.setFilterKeyColumn( 2 )
+        self.proxyPhotosModel.setFilterKeyColumn( self.PATH_COLUMN )
```

Comment:
This should be self.STATE_COLUMN, otherwise the combo box's activation will filter to an empty selection for all entries except "all".